### ISSUE TITLE
To merge or not to merge [part 2]

### ISSUE THREAD

**Issue by ghost** [wownero/meta#27]
The last time we had a [discussion about merge mining](https://github.com/wownero/meta/issues/3), the general consensus was to push the envelope by adopting RandomJS (RandomX) early in order to serve as an interesting test-bed for Monero. We forked to [RandomWOW on 2019-06-14](https://explore.wownero.com/search?value=114969) before any audits and looking back the [network has been relatively stable](https://stoffu.github.io/diff-chart/wownero.html). However, given how small the network is (usually 2-3 MH/s), it wouldn't take much effort or expense to attack it. So, should we reconsider merge mining with Monero?

Merge mining could help [secure the network by having a dominate chain with larger hash power](https://tlu.tarilabs.com/merged-mining/merged-mining-scene/MergedMiningIntroduction.html). Also, Monero's tail emission kicks in from 2022. Monero merge mining with [Tari](https://www.tari.com/faq/) + Wownero could provide [further revenue for miners to mine Monero even with the lower block reward](https://research.binance.com/analysis/merged-mining), which would be a win-win-win situation for all chains.

My initial hesitation with merge mining is that it would [encourage “hash & dump” behavior](
https://github.com/wownero/meta/issues/3#issuecomment-389138026), but even with a unique PoW,  this problem still occurs with profit switching mega pools. If there is going to be “hash & dump” behavior with or without an independent PoW, might as well have the security of mining with a dominate chain.

**Comment by qvqc** [comment]
I believe the uncertain hashrate has been a product of new interest and is growing pains for the coin.  I can say with out a doubt if wownero had been a merged mined coin my skepticism would have stopped me from being interested.  This is probably irrational.

randomwow has also been a huge selling point to friends who are interested in supporting a network that is more fair to its participants.  Most people I have known over the years were excited about currency being produced by computers but are disheartened by ASIC manufacture.  We need a not-so-shitty-shit-coin that we can mine while we are not playing games and we can give to our family and kids to think about how currency wowrks.   If this is dominated by ASICs it will feel more like some hashCorps company dollar than some dreamy future possibility.  

edit: initial thoughts were to people suggesting sha3 in irc..  I honestly would feel the same about something merged mine with monero, probably seeing wow as more of a joke than its own currency.

**Comment by SChernykh** [comment]
Pro: lots of hashrate
Cons:
- irremovable dependency on the other blockchain
- this hashrate is not dedicated and can vaporize any moment

**Comment by jwinterm** [comment]
> * irremovable dependency on the other blockchain

Is it really an irremovable dependency? If at a later date Wownero decided to fork to some other algorithm, whether that would be merge mined or not, then isn't the dependency on Monero blockchain removed at that point? You won't need to run Monero to validate the chain during the Monero merge-mined period, right?

**Comment by SChernykh** [comment]
Right, validators only need to check PoW, they don't need to verify if it actually came from Monero blockchain. Solo miners and pools still need to run both, but it's a small inconvenience.

**Comment by ghost** [comment]
I'm slightly in favor of SHA-3, seeing Aeon seems to be doing fine after switching to it. Lighter to verify on mobile devices.

**Comment by bomb-on** [comment]
> Solo miners and pools still need to run both, but it's a small inconvenience.

Pool operator here. If I would have to manage multiple blockchains on my pool, my expenses would be at least tripled immediately. This is not a small inconvenience imho... Running such a pool with a handful of miners, barely earning any fees, with current $WOW price simply won't be a viable option to me.
Just my 0.02 WOW...

**Comment by qvqc** [comment]
> Pool operator here. If I would have to manage multiple blockchains on my pool, my expenses would be at least tripled immediately. This is not a small inconvenience imho... 

Good point.  This would add to my costs as well.

**Comment by BugBud** [comment]
The difference between standalone mining and merge mining is the difference between having a self referential cryptocurrency with intrinsic value and a token built on the backbone of another chain. When Aristotle defined good money one property he identified for good money was that it must have "intrinsic value". This value of money should be independent of any other object and contained in the money itself. That is what allows a good crypto to make makes things commensurable because it makes it possible to equalize them. The way ratios for a fair exchange of heterogeneous things are set in crypto is through the delicate balance between hashrate, difficulty and value. Anything that disrupts that balance or obfuscates the real energy, hashrate and difficulty will inevitably do the same to the  intrinsic value in the network. This intrinsic value is analogous to the sovereignty of a nation, it is what allows our value to stand on its own merit instead of depending on another chain and being reactionary to whatever they do.  Another consideration is the the conflicting distribution schedule, WOW chain has a much longer distribution curve than Monero and the tail emission could affect our hashrate and value negatively down the line. We are not here to temporarily pump WOW with misleading hashrate only to have future users trying to figure out what to make of volatile hashrate shifts on a foreign chain. If we cannot measure the real hashrate against the value WOW wont be a unit that supplies a measure on the basis of which just exchange can take place. If we rely on existing Monero miners for our hashrate they will have no connection to WOW and will likely sell it. The fact that WOW miners choose to mine WOW for its own merits and not Monero is important to building a strong community and rewarding fairly the people that care about WOW instead of a bunch of wallflowers with no interest in advancing our coin. As for the security implications of such a move we might inherit the security of some Monero miners but at the end of the day that will give people a false impression of the security of our coin when we have to stand on our own to mine without monero possibly very soon depending on how the tail emission plays out. If there is one truth in this world, it's that WOW periodically claims it'll help us secure livelihoods and a better world for all generations. Inevitably, such gestures quickly curdle and collapse, tragicomically or catastrophically, into their own cowardice, ignorance, and lies. All we can do at that point is sit back and contemplate how we have to laugh when WOW says that going through the motions of mining is the same as mining. Where in the world did we get that idea? Not only does that idea contain absolutely no substance whatsoever, but unless you define success using the sort of loosey-goosey standards by which it abides you'll realize that true measures of success involve preventing the “invisible hand” of loosely regulated markets from becoming an “invisible fist” that lets WOW effortlessly pound its foes into oblivion. Success is getting the world to see that WOW contends that we should create an ideological climate that will enable it to maintain social control by eliminating miners rights and freedoms. The truth is that we are better than that. The truth is that the one thing that's central to all of WOW's self-indulgent endeavors is a desire to substitute rumor and gossip for bona fide evidence. I call this the New Cronyism. The old cronyism was concerned only with institutionalizing savagism through systematic violence, distorted religion, and dubious science. Although that was bad enough, we need to maintain independent mining. Why? Because of what's at stake: literally everything. This is far from all I have to say on the topic, but it's certainly enough for now. Just remember one thing:  Merge mined tokens use a twisted and self-contradictory logic to arrive at their value, just like premined coins, POS coins and centrally distributed ICOs  before them. Blockchain is life, WOW is alive and you are trying to turn this beautiful little creature into a no good dirty parasite.

**Comment by jwinterm** [comment]
> Blockchain is life, WOW is alive and you are trying to turn this beautiful little creature into a no good dirty parasite.

Seems like general consensus so far is stand pat and maybe eventually switch to SHA3 or similar. Personally I'm on the fence. On the one hand merge mining has kept Doge relatively secure and widely accepted for years now, and maybe would offer Wownero a sneaky way to broader awareness and acceptance by the Monero community and other exchanges in the near term. On the other hand, I don't want to discount the costs that would be imposed on pool operators, the much higher difficulty to mine coins, and the thesis put forth by @BugBud.

**Comment by ghost** [comment]
> I'm slightly in favor of SHA-3, seeing Aeon seems to be doing fine after switching to it. Lighter to verify on mobile devices.

We could always reduce RandomWOW scrathpad size to 256 KB and increase program size to reduce dependency on RAM (assuming this would seed up verification time).

**Comment by ghost** [comment]
> We could always reduce RandomWOW scrathpad size to 256 KB and increase program size to reduce dependency on RAM (assuming this would seed up verification time).

Yes, I guess what I meant was trivially verifiable. RandomX works well for the dominating coin, but is kind of useless for everyone else. If we eventually want ASIC to help securing the network, might as well bring them on board sooner rather than later.

**Comment by qvqc** [comment]
Im against switching to ASICs for at least a few years. It is too early. It would take away easy mining from newly interested parties.  They bring too much production centralization and don't add security respective to power consumption.  

If RandomWOW optimizations could help speed up verification on lower end hardware that seems like a nice middle ground with out much risk.

**Comment by 00-matt** [comment]
> > Solo miners and pools still need to run both, but it's a small inconvenience.
> 
> Pool operator here. If I would have to manage multiple blockchains on my pool, my expenses would be at least tripled immediately. This is not a small inconvenience imho... Running such a pool with a handful of miners, barely earning any fees, with current $WOW price simply won't be a viable option to me.
> Just my 0.02 WOW...

You could get by using one of the public Monero nodes

**Comment by ghost** [comment]
> It would take away easy mining from newly interested parties.

If you look at the current distribution of hash rate, there are no interested parties. In fact, most miners mining wownero don't even know what they are mining. At least SHA-3 will push the design of ASIC which will bring a more environmental friendly future.

Actually these have been discussed to death in monero, I really can not add anything new here :D

**Comment by 00-matt** [comment]
> At least SHA-3 will push the design of ASIC which will bring a more environmental friendly future.

Mining ASICs aren't environmentally friendly. They waste power just the same as a CPU (the goal of PoW) but they turn into e-waste faster.

**Comment by cryptonote-social** [comment]
It's true that any rapidly adjusting difficulty algorithm incentivizes profit switching, and the game theoretic outcome is a stalled chain should the difficulty get bounced too high. We see that starting to play out with Wownero where block times are regularly 10x expectation. But we all know that slowly adjusting difficulty puts smaller chains at grave risk of 51% attacks. Switching to ASIC friendly algo only makes sense if it's a unique algo not already in use by a larger coin, otherwise these mechanics remain.

Merge mining is a great way to secure a smaller chain, but IMO it makes more sense for chains that add functionality to the main chain (e.g. Namecoin, Tari) rather than (mostly) duplicate it. So unless Wownero adds some complementary features to Monero, I'm not sure I'm in favor of it.

Another solution could be some Proof-of-Stake like component, but that's got a whole other set of issues. I wonder if it's feasible to to allow for a Proof-of-Stake consensus to kick in should miners fail to generate a PoW block after some amount of time? Then you could use a fast difficulty adjusment without risk of a stalled chain. I would guess timestamp forging could make this hard/impossible?

**Comment by ghost** [comment]
> Switching to ASIC friendly algo only makes sense if it's a unique algo not already in use by a larger coin, otherwise these mechanics remain.

If we switch to something like KangarooTwelve, AFAIK Aeon is the first and only CryptoNote project using SHA-3. However, a unique algo does not necessary alleviate [long block times](https://stoffu.github.io/diff-chart/aeon-block-time-1.html) resulting from [hash attacks](https://stoffu.github.io/diff-chart/aeon-1.html) (checkout 7th April).

> I wonder if it's feasible to to allow for a Proof-of-Stake consensus to kick in should miners fail to generate a PoW block after some amount of time?

I am not really a fan of PoS. Loki uses [service nodes for checkpointing quorums](https://github.com/loki-project/loki-core/issues/620), but IMO this introduces new attack vectors. Monero already has an [emergency checkpointing system](https://monerodocs.org/infrastructure/monero-pulse/) in place, which we could set up for Wownero.

**Comment by jwinterm** [comment]
Zano also has a more traditional hybrid PoW/PoS system that is cryptonote
based:
https://zano.org/downloads/zano_wp.pdf

But I think they are pretty far diverged from Monero codebase, and yea I am
also not really a big PoS fan.

**Comment by BugBud** [comment]
The only reason I buy Wownero is to support my Junkie Brother Jeff and keep him off my couch. Jeff has 20 - 30 computers he stole from our old high school hes been using to mine Wownero and heat his apartment. Hope for the Wownero price is all that keeps Jeff going. He tells our mother hes a big shot professional miner. Jeff cannot Stake coins he needs to sell them to cover "costs" nor can Jeff buy a SHA-3 ASIC. If you adopt any of the proposed measures my junked out brother will end up on my couch again. What we need are more miners, more individual gamers, graphic designers, admins and junkies with hoards of stolen computers to mine.  That is how we secure not only the blockchain but all the Jeffs of the world. The possible upside of mining WOW is bigger than almost any other coin for the vast majority of people with little to invest in an ICO or staked coin, no specialized hardware and little experience. Get rid of that and you threaten the future of our coin more than any 51% attack. One way to combat profit switching pools is to attract dedicated miners and maintainers, that dedication to a single chain pays higher dividends to all involved than any profit switching algorithm ever will. Our dedication to general purpose machines means that machines being used today in data centers can be switched to mining WOW when idle. There is more hashpower sitting idle in data centers right now than the combined hashpower of every CPU profit switching pool in existence. A strategy that attracts those resources wont only secure the network possibly more than Monero one day. It will give each miner an unmatched opportunity to participate in a widely distributed long emission cryptocurrency while creating added incentives for every data center, admin and ecommerce site to integrate WOW payments if their already mining it with idle resources. The alternative is having Jihan WOW somewhere mining all the coins and trolling newcomers with overpriced ASICS. Ive also heard that Tezos, NEO and TRON are planning to vote on fully integrating WOWs codebase into their POS coins so I think we are pretty late on the ball for any of that.

**Comment by ghost** [comment]
We can combat whatever and keep this horribly slow way to validate blocks and use "--fast-sync" on mobile forever, or be as light as Aeon and always enable block verification starting from the hardfork.

Every mobile user can thank you, including Jeff, since he will likely mine Monero or better yet use one of the profit switching pools to accumulate Wow faster than ever, yet still keep his cellphone cooled while syncing the daemon, with the security he deserved.

**Comment by BugBud** [comment]
It is not about being light or mobile it is about a more fairly mined coin that gives anyone an opportunity to mine the chain with general purpose hardware. There will soon be hundreds of merge mined tokens on Monero ( tari )and its critical for us to have an actual chain to stand on our own feet and not be dependent on the Monero chain. It would turn WOW into a token. Having WOW on Jeffs phone does not not help Jeff he has to be able to mine them and frankly if Jeff has to compete with every Monero miner he simply wouldnt do it. WOW is not a test bed for SHA-3, literally everyone would just abandon it because were not getting SHA-3 miners and were not buying or holding a coin we cant mine. We will get attacked by whoever owns a large number of SHA-3 ASICs before any profit switching pool anyway.

**Comment by jwinterm** [comment]
>It is not about being light or mobile it is about a more fairly mined coin that gives anyone an opportunity to mine the chain with general purpose hardware. 

There is a tradeoff here: keeping it mineable by CPUs using RandomWhatever eventually makes it much more onerous for anyone with general purpose hardware to sync and validate the chain. SHA3 doesn't need a testbed - it is adopted by NIST as the standard third generation hash algorithm. I guess my ideal scenario is that some person that likes WOW is very rich and will make initial run of SHA3 ASICs and distribute them below cost...or something. I dunno.

**Comment by BugBud** [comment]
Well when you get the Bill and Melinda Gates foundation to make you and everyone else some ASICS I might reconsider in the meantime lets be realistic not everyone needs to sync and validate the chain. They can run remote nodes and its not like they will need ASICS to sync the chain. Its not a fair comparison in one case every miner needs an ASIC dedicated to one task and in another users who dont want to use a remote node will need more resources on still a general purpose machine to run full nodes. Its not like ull need an ASIC to sync the chain if we go with general purpose machines. Memory and processors will only get better over time so future cell phones will likely be able to run full nodes anyway.

**Comment by jwinterm** [comment]
pastebin of PoW conversation on freenode:  
https://pastebin.com/Nwx0bZeK

tl;dr:   
maybe random-wow way turned down but change scratchpad to use blockchain data to eliminate pooled mining?

**Comment by ghost** [comment]
Did Satoshi care about the opinions from investors?

**Comment by 00-matt** [comment]
jw wrote: 
> tl;dr:
> maybe random-wow way turned down but change scratchpad to use 
> blockchain data to eliminate pooled mining?

Wow, a blessed idea! What blockchain data do we use though? If it was only a small piece per block template, pools could just send it to miners. If it's large and includes parts from all over the chain, people using hard drives will have a very bad time.

***

fuwa wrote:
> Did Satoshi care about the opinions from investors?

Satoshi never innovated in the field of meme based currencies, and also he disappeared and does not matter anymore.

**Comment by cryptonote-social** [comment]
If you want to go the route of using blockchain data to eliminate pooled mining,  Nerva may contain some useful ideas+experience to consider.

**Comment by tevador** [comment]
An alternative to RandomX with fast verification: https://github.com/tevador/hashx

**Comment by cryptonote-social** [comment]
> An alternative to RandomX with fast verification: https://github.com/tevador/hashx

cool, tell us more!  Looks like it's integer operations only -- does that mean it's going to be less specialized-hardware-proof than RandomX?

**Comment by tevador** [comment]
> does that mean it's going to be less specialized-hardware-proof

It may be, but no floating point ops make it easier to implement and review and there are fewer chances of something breaking. We had to jump through a lot of hoops to make floating point work in RandomX.

Also HashX has no branches, because it's impossible to make them work while keeping the runtime constant for all possible seeds.

Overall, HashX requires a simpler CPU than RandomX, but otherwise shares a similar specialized hardware resistance profile (apart from the memory requirements).

**Comment by tevador** [comment]
https://github.com/tevador/equix
https://github.com/tevador/equix/blob/master/devlog.md

The main purpose of this algorithm is to be a lightweight client puzzle for DoS protection, but I guess it could also be used by Wownero and replace RandomX.

**Comment by ghost** [comment]
migrated to https://git.wownero.com/wownero/meta
